inc/class-s3-uploads.php

```diff
@@ -30,11 +30,14 @@ public function __construct( $bucket, $key, $secret, $bucket_hostname = null ) {
 		$this->secret = $secret;
 		$this->bucket_hostname = $bucket_hostname ? '//' . $bucket_hostname : 'https://' . strtok( $this->bucket, '/' ) . '.s3.amazonaws.com';
 
-		$this->s3()->registerStreamWrapper();
-		stream_context_set_option( stream_context_get_default(), 's3', 'seekable', true );
-		stream_context_set_option( stream_context_get_default(), 's3', 'ACL', Aws\S3\Enum\CannedAcl::PUBLIC_READ );
-		
-
+		if ( defined( 'S3_UPLOADS_USE_LOCAL' ) && S3_UPLOADS_USE_LOCAL ) {
+			require_once dirname( __FILE__ ) . '/class-s3-uploads-local-stream-wrapper.php';
+			stream_wrapper_register( 's3', 'S3_Uploads_Local_Stream_Wrapper', STREAM_IS_URL );
+		} else {
+			$this->s3()->registerStreamWrapper();
+			stream_context_set_option( stream_context_get_default(), 's3', 'seekable', true );
+			stream_context_set_option( stream_context_get_default(), 's3', 'ACL', Aws\S3\Enum\CannedAcl::PUBLIC_READ );
+		}
```

Comment:
We should set these context options always, and only do the minimal amount of difference.